mobx_codegen/lib/src/mobx_codegen_base.dart

```diff
@@ -59,69 +62,108 @@ class StoreMixinVisitor extends SimpleElementVisitor {
 
   StoreTemplate _storeTemplate;
 
-  String get source => _storeTemplate.toString();
+  final StoreClassCodegenErrors _errors;
+
+  final void Function(String message) _logSevere;
+
+  String get source {
+    if (_errors.hasErrors) {
+      _logSevere(_errors.message);
+      return '';
+    }
+    return _storeTemplate.toString();
+  }
 
   @override
   visitFieldElement(FieldElement element) async {
-    if (!element.isFinal && _observableChecker.hasAnnotationOfExact(element)) {
-      final template = ObservableTemplate()
-        ..storeTemplate = _storeTemplate
-        ..atomName = '_\$${element.name}Atom'
-        ..type = element.type.displayName
-        ..name = element.name;
-
-      _storeTemplate.observables.add(template);
+    if (!_observableChecker.hasAnnotationOfExact(element)) {
+      return null;
     }
+
+    if (_checkFieldErrors(element)) {
```

Comment:
I would recommend a name like `hasFieldErrors` or in general `hasXXXErrors` so it reads more like english ;-).